test/integration/targets/aci_interface_policy_leaf_policy_group/tasks/main.yml

```diff
@@ -0,0 +1,361 @@
+# Test code for the ACI modules
+# Copyright 2017, Bruno Calogero <bcaloger@cisco.com>
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI APIC host, ACI username and ACI password
+  fail:
+    msg: 'Please define the following variables: aci_hostname, aci_username and aci_password.'
+  when: aci_hostname is not defined or aci_username is not defined or aci_password is not defined
+
+- name: Making sure interface_policy_leaf_policy_group doesn't exist at beggining of test (PC)
+  aci_interface_policy_leaf_policy_group: &aci_interface_policy_leaf_policy_group_link_absent
+    host: "{{ aci_hostname }}"
+    username: "{{ aci_username }}"
+    password: "{{ aci_password }}"
+    policy_group: policygroupname
+    lag_type: link
+    state: absent
+
+- name: Making sure interface_policy_leaf_policy_group doesn't exist at beggining of test (VPC)
+  aci_interface_policy_leaf_policy_group: &aci_interface_policy_leaf_policy_group_node_absent
+    host: "{{ aci_hostname }}"
+    username: "{{ aci_username }}"
+    password: "{{ aci_password }}"
+    policy_group: policygroupname
+    lag_type: node
+    state: absent
+
+- name: Making sure interface_policy_leaf_policy_group doesn't exist at beggining of test (Leaf Access Port)
+  aci_interface_policy_leaf_policy_group: &aci_interface_policy_leaf_policy_group_leaf_absent
+    host: "{{ aci_hostname }}"
+    username: "{{ aci_username }}"
+    password: "{{ aci_password }}"
+    policy_group: policygroupname
+    lag_type: leaf
+    state: absent
+
+
+# ==== TESTING Port Channel (PC), lag_type: link ====
```

Comment:
I would split these different tests up into multiple playbooks that are included in the main.yml.

Look at the tests for **aci_encap_pool**.

Reply:
Still valid, but I am going to merge this. We can review this later.